### PR TITLE
#19143 Implement render field at page level. Postman test

### PR DESCRIPTION
**Pull request by dsilvam**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotcms.rendering.velocity.servlet.VelocityModeHandler;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:9]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.VelocityUtil;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.io.CharArrayReader;
+import java.io.IOException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:27]: Codacy found an issue: [Avoid unused imports such as 'java.io.IOException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotcms.rendering.velocity.servlet.VelocityModeHandler;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotcms.rendering.velocity.servlet.VelocityModeHandler;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:7]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.business.PermissionLevel'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRaw;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRenderedBuilder;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:15]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRenderedBuilder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            final PageMode mode = PageMode.get(pageModeAsString);
+
+            final HTMLPageAsset pageAsset = APILocator.getHTMLPageAssetAPI()
+                    .fromContentlet(contentlet);
+
+            final Host host = WebAPILocator.getHostWebAPI().getCurrentHost(request, user);
+
+            return VelocityModeHandler.modeHandler(pageAsset, mode, request, response, host).eval();
+
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:62]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRaw;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRenderedBuilder;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.JsonMapper;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.templates.model.Template;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:18]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.templates.model.Template'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.VelocityUtil;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.util.VelocityUtil'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.util.Collection;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:28]: Codacy found an issue: [Avoid unused imports such as 'java.util.Collection'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRaw;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRenderedBuilder;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.JsonMapper;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.VelocityUtil;
+import com.fasterxml.jackson.databind.ObjectWriter;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:22]: Codacy found an issue: [Avoid unused imports such as 'com.fasterxml.jackson.databind.ObjectWriter'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotcms.rendering.velocity.servlet.VelocityModeHandler;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 12]
```diff
@@ -0,0 +1,67 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotcms.rendering.velocity.servlet.VelocityModeHandler;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:10]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotRuntimeException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRaw;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRenderedBuilder;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.JsonMapper;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:16]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.page.JsonMapper'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.util.VelocityUtil;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:30]: Codacy found an issue: [Avoid unused imports such as 'java.util.Objects'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.velocity.context.Context;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:33]: Codacy found an issue: [Avoid unused imports such as 'org.apache.velocity.context.Context'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.VelocityUtil;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:29]: Codacy found an issue: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRaw;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:14]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRaw'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.VelocityUtil;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.io.CharArrayReader;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:26]: Codacy found an issue: [Avoid unused imports such as 'java.io.CharArrayReader'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+                    != null ? environment.getArgument("pageMode") : PageMode.LIVE.name();
+
+            final PageMode mode = PageMode.get(pageModeAsString);
+
+            final HTMLPageAsset pageAsset = APILocator.getHTMLPageAssetAPI()
+                    .fromContentlet(contentlet);
+
+            final Host host = WebAPILocator.getHostWebAPI().getCurrentHost(request, user);
+
+            return VelocityModeHandler.modeHandler(pageAsset, mode, request, response, host).eval();
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:60]: here you can use this better

https://github.com/dotCMS/core/blob/bca720b97f5ed7fc9e5d76aa58d2e51c47b25bd8/dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java#L464

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:30]: Codacy found an issue: [Avoid unused imports such as 'java.util.Objects'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:9]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotDataException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:15]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRenderedBuilder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 17 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:14]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.ContainerRaw'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:10]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotRuntimeException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 7 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:18]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.templates.model.Template'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:7]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.business.PermissionLevel'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:29]: Codacy found an issue: [Avoid unused imports such as 'java.util.Map'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 18 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:26]: Codacy found an issue: [Avoid unused imports such as 'java.io.CharArrayReader'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:27]: Codacy found an issue: [Avoid unused imports such as 'java.io.IOException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 8 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:21]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.util.VelocityUtil'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:16]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.render.page.JsonMapper'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 6 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:62]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 9 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:28]: Codacy found an issue: [Avoid unused imports such as 'java.util.Collection'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:11]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:33]: Codacy found an issue: [Avoid unused imports such as 'org.apache.velocity.context.Context'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:22]: Codacy found an issue: [Avoid unused imports such as 'com.fasterxml.jackson.databind.ObjectWriter'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:13]: Codacy found an issue: [Avoid unused imports such as 'com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java (hunk 19 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java]: Done

File: dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+
+            final PageContext pageContext = PageContextBuilder.builder()
+                    .setUser(user)
+                    .setPageUri(url)
+                    .setPageMode(mode)
+                    .build();
+
+            return APILocator.getHTMLPageAssetRenderedAPI().getPageHtml(pageContext, request,
+                    response);
+        } catch (Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/PageRenderDataFetcher.java:44]: Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6063675)

**Review by victoralfaro-dotcms — APPROVED** [review]
LGTM

**Pull request merged**